### PR TITLE
Add "Nextcloud" extension

### PR DESCRIPTION
**Pull request by uetchy**
## Description

I added a new extension to search files, task boards, notes, favorites, and activities in [Nextcloud](https://nextcloud.com/).

## Screencast

Screenshot1

Screenshot2

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `media` folder

### CODE REVIEW

File: extensions/nextcloud/src/search/hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  return {
+    search,
+    isLoading,
+    results: data ?? [],
+  };
+}
+
+function makeBodyForSearch({ username, query, scope = "" }: { username: string; query: string; scope?: string }) {
+  return `<?xml version="1.0" encoding="UTF-8"?>
+<d:searchrequest xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
```
Comment (pernielsentikaer) [inline extensions/nextcloud/src/search/hooks.tsx:24]:
```suggestion
<d:searchrequest xmlns:d="DAV:" xmlns:oc="https://owncloud.org/ns">
```
Shouldn't this be on a secure connection?

File: extensions/nextcloud/src/favorites/hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      dirname,
+      filename,
+    };
+  });
+  return res;
+}
+
+function makeBodyForListFavorites() {
+  return `<?xml version="1.0"?>
+    <oc:filter-files  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns">
```
Comment (pernielsentikaer) [inline extensions/nextcloud/src/favorites/hooks.tsx:37]:
```suggestion
    <oc:filter-files  xmlns:d="DAV:" xmlns:oc="https://owncloud.org/ns" xmlns:nc="https://nextcloud.org/ns">
```
Shouldn't this be on a secure connection?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @uetchy 👋 

Thanks for a great contribution, do you by any chance have the possibility to add me to an owncloud account (scoped) so I don't have access to anything?

If not, it's also okay – then you maybe create a screencast showing it quickly to get a bit more feeling 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by uetchy** [comment]
Hi @pernielsentikaer
You can sign up for a free and testing purpose Nextcloud account from several providers via <https://nextcloud.com/signup/>👍

File: extensions/nextcloud/src/search/hooks.tsx (hunk 1 above)
Comment (uetchy) [reply extensions/nextcloud/src/search/hooks.tsx:24]: They are taken from [the official API documentation](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/WebDAV/basic.html), and I'm not so sure about changing the XML namespace schema address as some servers may require that URL particularly be of `http://...`.

**Review by pernielsentikaer — APPROVED** [review]
Hi @uetchy 👋 

Thanks for the clarification, now it's possible to test, thanks! 🔥 

I'm going to approve it, looking good to me 💪

**Pull request merged**